[BUG] Test Failure org.opensearch.index.engine.NRTReplicationEngineTests.testUpdateSegments

**Describe the bug**
Comes from https://github.com/opensearch-project/OpenSearch/pull/3568#issuecomment-1154722879

**To Reproduce**
```
> Task :server:test

REPRODUCE WITH: ./gradlew ':server:test' --tests "org.opensearch.index.engine.NRTReplicationEngineTests.testUpdateSegments" -Dtests.seed=569CC7C1CA942529 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=sr-RS -Dtests.timezone=Africa/Lome -Druntime.java=17

org.opensearch.index.engine.NRTReplicationEngineTests > testUpdateSegments FAILED
    java.lang.AssertionError: expected:<[_0.cfe, _0.si, _0_1_Asserting_0.dvm, _2.si, _0.cfs, _2.cfe, _2.cfs, _0_1_Asserting_0.dvd, _0_1.fnm, segments_3]> but was:<[_0.si, _0_1_Asserting_0.dvm, _1.cfe, _1.si, _0.cfs, _0_1_Asserting_0.dvd, segments_3, _1.cfs, _1_1_Asserting_0.dvd, _0.cfe, _1_1_Asserting_0.dvm, _0_1.fnm, _1_1.fnm]>
        at __randomizedtesting.SeedInfo.seed([569CC7C1CA942529:36DB290B658B5CC0]:0)
        at org.junit.Assert.fail(Assert.java:89)
        at org.junit.Assert.failNotEquals(Assert.java:835)
        at org.junit.Assert.assertEquals(Assert.java:120)
        at org.junit.Assert.assertEquals(Assert.java:146)
        at org.opensearch.index.engine.NRTReplicationEngineTests.assertMatchingSegmentsAndCheckpoints(NRTReplicationEngineTests.java:203)
        at org.opensearch.index.engine.NRTReplicationEngineTests.testUpdateSegments(NRTReplicationEngineTests.java:143)
```

**Host/Environment (please complete the following information):**
 - OS: [e.g. iOS]
 - Version [e.g. 22]

**Additional context**
Add any other context about the problem here.

--- a/server/src/test/java/org/opensearch/index/engine/NRTReplicationEngineTests.java
+++ b/server/src/test/java/org/opensearch/index/engine/NRTReplicationEngineTests.java
@@ -110,8 +110,9 @@ public class NRTReplicationEngineTests extends EngineTestCase {
 
             engine.refresh("test");
 
-            nrtEngine.updateSegments(engine.getLatestSegmentInfos(), engine.getProcessedLocalCheckpoint());
-            assertMatchingSegmentsAndCheckpoints(nrtEngine);
+            final SegmentInfos latestPrimaryInfos = engine.getLatestSegmentInfos();
+            nrtEngine.updateSegments(latestPrimaryInfos, engine.getProcessedLocalCheckpoint());
+            assertMatchingSegmentsAndCheckpoints(nrtEngine, latestPrimaryInfos);
 
             // assert a doc from the operations exists.
             final ParsedDocument parsedDoc = createParsedDoc(operations.stream().findFirst().get().id(), null);
@@ -139,8 +140,9 @@ public class NRTReplicationEngineTests extends EngineTestCase {
                 );
             }
 
-            nrtEngine.updateSegments(engine.getLastCommittedSegmentInfos(), engine.getProcessedLocalCheckpoint());
-            assertMatchingSegmentsAndCheckpoints(nrtEngine);
+            final SegmentInfos primaryInfos = engine.getLastCommittedSegmentInfos();
+            nrtEngine.updateSegments(primaryInfos, engine.getProcessedLocalCheckpoint());
+            assertMatchingSegmentsAndCheckpoints(nrtEngine, primaryInfos);
 
             assertEquals(
                 nrtEngine.getTranslog().getGeneration().translogFileGeneration,
@@ -196,14 +198,14 @@ public class NRTReplicationEngineTests extends EngineTestCase {
         }
     }
 
-    private void assertMatchingSegmentsAndCheckpoints(NRTReplicationEngine nrtEngine) throws IOException {
+    private void assertMatchingSegmentsAndCheckpoints(NRTReplicationEngine nrtEngine, SegmentInfos expectedSegmentInfos)
+        throws IOException {
         assertEquals(engine.getPersistedLocalCheckpoint(), nrtEngine.getPersistedLocalCheckpoint());
         assertEquals(engine.getProcessedLocalCheckpoint(), nrtEngine.getProcessedLocalCheckpoint());
         assertEquals(engine.getLocalCheckpointTracker().getMaxSeqNo(), nrtEngine.getLocalCheckpointTracker().getMaxSeqNo());
-        assertEquals(engine.getLatestSegmentInfos().files(true), nrtEngine.getLatestSegmentInfos().files(true));
-        assertEquals(engine.getLatestSegmentInfos().getUserData(), nrtEngine.getLatestSegmentInfos().getUserData());
-        assertEquals(engine.getLatestSegmentInfos().getVersion(), nrtEngine.getLatestSegmentInfos().getVersion());
-        assertEquals(engine.segments(true), nrtEngine.segments(true));
+        assertEquals(expectedSegmentInfos.files(true), nrtEngine.getLatestSegmentInfos().files(true));
+        assertEquals(expectedSegmentInfos.getUserData(), nrtEngine.getLatestSegmentInfos().getUserData());
+        assertEquals(expectedSegmentInfos.getVersion(), nrtEngine.getLatestSegmentInfos().getVersion());
     }
 
     private void assertSearcherHits(Engine engine, int hits) {

